lib/ABI/FunctionType.cpp

```diff
@@ -169,24 +175,31 @@ class ConversionHelper {
 
     model::RawFunctionType Result;
     Result.CustomName = Function.CustomName;
```

Comment:
As a matter of a fact, we do not. But neither are we adding this function to the model `;)`. It's up to the user what to do with the returned `model::RawFunctionType` object.

Reply:
Mmmmh, we're not adding it to the model, but we're adding other `StructType`s! Either we add everything or we don't add anything.
There should be only one subject responsible for adding all the things to the model. If it can't be `toRaw`'s caller, it has to be `toRaw` itself.

Reply:
But if we're adding resulting function to the model, I'd push for removal of the original one and replacement of all the references. But at that point it feels like this function is doing way too much "damage" to the original model.

Reply:
None than I can think of. In all honesty, that's what bulk version (the `revng-model-opt` pass) is doing already.

I have no practical opposition to moving it all higher up. I do feel like it's wrong, but whatever.

Reply:
I think we want `toRaw` inject the new prototype in the model.
Then, we could have another function that exposes an internal functionality, but you would need to call it `convertToRawButDontAddToModelButAddSomeOtherStuffToTheModel`.

Reply:
Sounds good to me.

Reply:
Done: https://github.com/revng/revng/pull/148/commits/bdda96b74c961763ebbab381626c4eed68cd353e.